### PR TITLE
Implement `dropIndexes` command

### LINKED ISSUE #1512: Implement `dropIndexes` command
Depends on https://github.com/FerretDB/FerretDB/issues/1509

https://www.mongodb.com/docs/manual/reference/command/dropIndexes/

DoD:
- [x] both `dropIndex` and `dropIndexes` (`dropOne` and `dropAll`)
- [x] deletion by index name (also, multiple names for `dropIndexes`)
- [x] deletion by index spec 
- [x] deletion by `*`
- [x] check that `_id` index can't be deleted
- [x] update supported commands docs
- [x] test parallel deletion of indexes

Not in scope:
- `comment`

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1512.

## Readiness checklist

* [x] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — APPROVED** [review]
I think we did a great job working on this PR 🤓 

I created a separate issue https://github.com/FerretDB/FerretDB/issues/2337 to cover some minor things that are left

**Review by Fashander — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]
Great job! Changes look consistent to me!

**Review by noisersup — APPROVED** [review]
Huge work! LGTM

**Review by AlekSi — APPROVED** [review]

File: internal/handlers/pg/pgdb/indexes.go [hunk 1]
```diff
@@ -80,6 +80,103 @@ func CreateIndexIfNotExists(ctx context.Context, tx pgx.Tx, db, collection strin
...
+		if metadata.indexes[i].Name == "_id_" {
+			continue
+		}
+
+		if err = dropPgIndex(ctx, tx, db, metadata.indexes[i].pgIndex); err != nil {
+			return 0, lazyerrors.Error(err)
+		}
+
+		// remove i-th element from the slice
+		metadata.indexes = append(metadata.indexes[:i], metadata.indexes[i+1:]...)
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/indexes.go:170]: https://pkg.go.dev/golang.org/x/exp/slices#Delete

File: internal/handlers/pg/pgdb/indexes_test.go [hunk 2]
```diff
@@ -63,3 +66,295 @@ func TestCreateIndexIfNotExists(t *testing.T) {
...
+				was, err = DropAllIndexes(ctx, tx, databaseName, collectionName)
+				if err != nil {
+					return err
+				}
+
+				assert.Equal(t, expectedWas, was)
+
+				var indexes []Index
+				indexes, err = Indexes(ctx, tx, databaseName, collectionName)
+				if err != nil {
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/indexes_test.go:254]:
```suggestion
				if indexes, err = Indexes(ctx, tx, databaseName, collectionName); err != nil {
```

**Pull request merged**